autoscale/src/main/java/com/sequenceiq/periscope/service/PeriscopeMetricService.java

```diff
@@ -32,4 +39,26 @@ protected void init() {
     protected Optional<String> getMetricPrefix() {
         return Optional.of(METRIC_PREFIX);
     }
+
+    @Override
+    protected boolean gaugeMetric(Metric metric) {
+        return metric.getMetricName().contains("state") || metric.getMetricName().contains("leader") || metric.getMetricName().contains("threadpool")
+                || metric.getMetricName().contains("scaling.activity");
+    }
+
+    private String[] createTags(Cluster cluster) {
+        return new String[]{MetricTag.TENANT.name(), Optional.ofNullable(cluster.getClusterPertain().getTenant()).orElse("NA"),
```

Comment:
Can cluster.getClusterPertain() be null?

Reply:
We had seen issues previously where the ClusterPertain entity did not have correct unique constraints when being persisted into the DB. I added a null check to be safe. Also, I followed the pattern on adding such tags for metrics from the core service

Reply:
Are we adding the null check on it? Can be another one which goes into the "NA" category ?

Reply:
Added an explicit null check